app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -226,7 +233,72 @@ enum SectionHeaderPlacement {
 const SECTION_HEADER_PLACEMENT: SectionHeaderPlacement = SectionHeaderPlacement::Bottom;
 
 
+
+// ========================
 // === Shape Definition ===
+// ========================
+
+
+// === Selection ===
+
+/// A shape of a selection box. It is used as a mask to show only specific parts of the selection
+/// layers. See [component_group](component_group#selection) module documentation to learn how this
+/// mask is used.
+///
+/// The selection `mask` is a rectangle with rounded corners. The final shape is an intersection of
+/// the `mask` and the `area` shape that covers the entire component list panel and has the same
+/// shape. This is needed to ensure that the selection box never leaves the borders of the
+/// component list panel. Instead of using `set_position` API methods of the display object, we
+/// have a `pos` parameter instead. Similarly, we have the `selection_size` parameter for the
+/// size of the `mask`.
+///
+/// The `area` shape consists of two parts and has a variable height. The top part is a simple
+/// rectangle, while the bottom part is a rounded rectangle, to repeat the rounded corners on the
+/// bottom of the component browser. The variable height is used to create an effect of the
+/// covering the selection with the component group's header. When the header is selected,
+/// `margin_top` is set to 0, so the `area` covers the whole component list panel and the
+/// selection is displayed on top of the header. When the header is not selected, the `area`'s
+/// height is reduced by the `margin_top`, and the selection is displayed "below" the header. We
+/// can't change the actual order of rendering here, but we clip the selection box so that it
+/// looks like the header covers the part of it.
+///
+/// We have only a single `corners_radius` parameter that is used both for the bottom corners of the
+/// `area` and the corners of the selection `mask` itself. In the current design, they are equal
```

Comment:
- Why the word "itself" is needed here?
- "in the current design" - no link, no one knows what this sentence refers to.
- "because the header never touches the bottom of the component list panel and there is no way to see the difference in the radius of the corner." - no explanation agani. I don't understand what this sentence is trying to tell me.

Reply:
The whole section is rewritten for clarity.